### ISSUE TITLE
Fully-transparent `z_sendmany` does not recommend `AllowFullyTransparent`

### ISSUE THREAD

**Issue by str4d** [zcash/zcash#6283]
If I call `z_sendmany TADDR1 {"recipient": TADDR2, ...} 1 0 AllowRevealedSenders`, the error message says that `AllowRevealedRecipients` is required. But if I replace the policy with `AllowRevealedRecipients`, I get another error message saying that `AllowRevealedSenders` is required. `z_sendmany` should instead recognise that the required policy is `AllowFullyTransparent`.

**Comment by str4d** [comment]
I encountered this while working on #6282. @sellout does your PR train fix this incidentally?

**Comment by sellout** [comment]
Yeah, it should – I ran into that exact failure case. https://github.com/zcash/zcash/pull/6217/commits/56064115f5d8dcf2d94ed8536c901f8b5e3fc9ca, in particular.

However, I think it still has a weakness – basically, it only accumulates the “weakenings” that the provided policy didn’t satisfy. From there, you have two options, either the thing you ran into, where we return the accumulated weakenings without taking the provided policy into account (so it can accidentally recommend a too-strong policy) or (what I do) `providedPolicy ∧ failedChecks`, which recommends a policy strictly weaker than the provided one (and possibly overly weak).

I think the _correct_ thing is to track _all_ of the weakenings that were required by the tx (whether or not the providedPolicy satisfied them) and then (if the provided policy is insufficiently weak) return that accumulation (which may not be comparable to the provided policy at all).

[helpful reference](https://github.com/zcash/zcash/issues/6240#issuecomment-1309622380)

To modify your example a bit, say you provided `AllowLinkingAccountAddresses`
* the current behavior would fail, recommending `AllowRevealedRecipients` (which is too strong)
* my patch, if you ignore the provided policy would do the same
* my patch, taking into account the provided policy would recommend `NoPrivacy`, because it’s the only thing below the provided policy and `AllowRevealedRecipients` (overly weak)
* the change I propose above would return `AllowFullyTransparent`, which is exactly the right policy, but it’s not comparable to the provided policy.
